modules/pricer/src/main/java/com/opengamma/strata/pricer/rate/bond/DiscountingFixedCouponBondTradePricer.java

```diff
@@ -0,0 +1,202 @@
+      DiscountingPaymentPricer paymentPricer) {
+    this.productPricer = ArgChecker.notNull(productPricer, "productPricer");
+    this.paymentPricer = ArgChecker.notNull(paymentPricer, "paymentPricer");
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the present value of the fixed coupon bond trade.
+   * <p>
+   * The present value of the trade is the value on the valuation date.
+   * The result is expressed using the payment currency of the bond.
+   * 
+   * @param trade  the trade to price
+   * @param provider  the rates provider
+   * @return the present value of the fixed coupon bond trade
+   */
+  public CurrencyAmount presentValue(FixedCouponBondTrade trade, LegalEntityDiscountingProvider provider) {
+    CurrencyAmount pvProduct = productPricer.presentValue(trade.getProduct(), provider);
+    return presentValueFromProductPresentValue(trade, provider, pvProduct);
+  }
+
+  /**
+   * Calculates the present value of the fixed coupon bond trade with z-spread.
+   * <p>
+   * The present value of the trade is the value on the valuation date.
+   * The result is expressed using the payment currency of the bond.
+   * <p>
+   * The z-spread is a parallel shift applied to continuously compounded rates or periodic
+   * compounded rates of the discounting curve.
+   * 
+   * @param trade  the trade to price
+   * @param provider  the rates provider
+   * @param zSpread  the z-spread
+   * @param periodic  if true, the spread is added to periodic compounded rates,
+   *  if false, the spread is added to continuously compounded rates
+   * @param periodsPerYear  the number of periods per year
+   * @return the present value of the fixed coupon bond trade
+   */
+  public CurrencyAmount presentValueWithZSpread(
+      FixedCouponBondTrade trade,
+      LegalEntityDiscountingProvider provider,
+      double zSpread,
+      boolean periodic,
+      int periodsPerYear) {
+    CurrencyAmount pvProduct =
+        productPricer.presentValueWithZSpread(trade.getProduct(), provider, zSpread, periodic, periodsPerYear);
+    return presentValueFromProductPresentValue(trade, provider, pvProduct);
+  }
+
+  /**
+   * Calculates the present value of the fixed coupon bond trade from the present value of the underlying fixed
+   * coupon bond product. 
+   * <p>
+   * It is assumed that the payment currency of the bond product is the same as the currency of the up-front payment of 
+   * the trade. 
+   * 
+   * @param trade  the trade to price
+   * @param provider  the rates provider
+   * @param productPresentValue  the present value of the product
+   * @return  the present value of the fixed coupon bond trade
+   */
+  public CurrencyAmount presentValueFromProductPresentValue(
+      FixedCouponBondTrade trade,
+      LegalEntityDiscountingProvider provider,
+      CurrencyAmount productPresentValue) {
+    CurrencyAmount pvProduct = productPresentValue.multipliedBy(trade.getQuantity());
```

Comment:
The product present value take into account all the coupons in the future down to today. The present value of the trade should include only the coupons that are after the settlement date/ ex-coupon date. An adjustment for those coupons is required in the present value.